src/connection-manager/index.ts

```diff
@@ -489,6 +505,13 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
       }
     }
 
+    if ((this.getConnections().length + 1) > this.opts.maxOutgoingConnections) {
```

Comment:
Shouldn't you filter `this.getConnections` to get only outbound connections?

Reply:
nit: we are already fetching all connections few lines above so it probably makes sense to reuse that array

Reply:
In the few lines above we are fetching all connections for a particular peerId as opposed to all open connections no?

Reply:
Ah, yeah, you are right!